gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/Contains.java

```diff
@@ -52,6 +52,21 @@ public enum Contains implements BiPredicate<Object, Collection> {
     within {
         @Override
         public boolean test(final Object first, final Collection second) {
+            if (first instanceof Element &&
+                    second instanceof BulkSet<?> &&
+                    ((BulkSet<?>)second).allContainedElementsSameClass() &&
+                    ((BulkSet<?>)second).getAllContainedElementsClass() != null &&
+                    Element.class.isAssignableFrom(((BulkSet<?>)second).getAllContainedElementsClass()) &&
```

Comment:
Could these checks be removed? I believe they are redundant. If any of these checks fail, then the final `first.getClass() == ((BulkSet<?>)second).getAllContainedElementsClass()` check would also fail.
```suggestion
```